client/asset/btc/btc.go

```diff
@@ -3349,8 +3349,17 @@ func convertUnspent(confs uint32, unspents []*ListUnspentResult, chainParams *ch
 
 			nfo, err := dexbtc.InputInfo(txout.ScriptPubKey, txout.RedeemScript, chainParams)
 			if err != nil {
+				if errors.Is(err, dex.UnsupportedScriptError) {
+					continue
+				}
 				return nil, nil, 0, fmt.Errorf("error reading asset info: %w", err)
 			}
+			if nfo.ScriptType == dexbtc.ScriptUnsupported || nfo.NonStandardScript {
```

Comment:
Can we check these internally to `InputInfo` and return a `dex.UnsupportedScriptError`?

It'll affect `*(*Backend).utxo` server-side, I'm wondering if we're handling unsupported script types properly there.

Reply:
> It'll affect `*(*Backend).utxo` server-side, I'm wondering if we're handling unsupported script types properly there.

That's the reason I opted to just mirror dcr.go instead of modifying `InputInfo`.  On server, the `(*Backend).utxo` method is designed to return info about all utxos, at least looking at the return:

```go
out := &Output{
	TXIO: TXIO{
		btc:        btc,
		tx:         tx,
		height:     blockHeight,
		blockHash:  blockHash,
		maturity:   int32(maturity),
		lastLookup: lastLookup,
	},
	vout:              vout,
	scriptType:        scriptType,
	nonStandardScript: inputNfo.NonStandardScript,
	pkScript:          pkScript,
	redeemScript:      redeemScript,
	numSigs:           inputNfo.ScriptAddrs.NRequired,
	spendSize:         inputNfo.VBytes(),
	value:             toSat(txOut.Value),
}
```

Specifically, that `nonStandardScript: inputNfo.NonStandardScript` part.

Maybe it should/could be changed, but I opted for the path of least resistance with the least likelihood of unintended consequences. 

Still worth investigating though.

Reply:
Ah, yeah, it requires non-standard to be not an error because of `(*Backend) Contract(coinID []byte, redeemScript []byte)` -> `(*Backend).output` -> `InputInfo(pkScript, redeemScript)`, where `redeemScript` is the swap contract (non-standard)

Reply:
Oh right.